tour_model_eval/label_processing.py

```diff
@@ -1,65 +1,77 @@
 import logging
-from scipy.cluster.hierarchy import linkage, dendrogram,fcluster
-
-
+import scipy.cluster.hierarchy as sch
 
 # to map the user labels
 # - user_input_df: pass in original user input dataframe, return changed user input dataframe
 # - sp2en: change Spanish to English
-# - cvt_pur_mo: convert purposes and replaced mode
-def map_labels(user_input_df,sp2en,cvt_pur_mo):
+def map_labels_sp2en(user_input_df):
     # Spanish words to English
     span_eng_dict = {'revisado_bike': 'test ride with bike', 'placas_de carro': 'car plates', 'aseguranza': 'insurance',
                      'iglesia': 'church', 'curso': 'course',
                      'mi_hija recién aliviada': 'my daughter just had a new baby',
                      'servicio_comunitario': 'community service', 'pago_de aseguranza': 'insurance payment',
                      'grupo_comunitario': 'community group', 'caminata_comunitaria': 'community walk'}
 
+    # change language
+    user_input_df = user_input_df.replace(span_eng_dict)
+    return user_input_df
+
+
+# to map purposes and replaced mode in user inputs
+# - cvt_pur_mo: convert purposes and replaced mode
+def map_labels_pur_mo(user_input_df):
     # Convert purpose
     map_pur_dict = {'course': 'school', 'work_- lunch break': 'lunch_break', 'on_the way home': 'home',
                     'insurance_payment': 'insurance'}
 
-    if sp2en:
-        # change language
-        user_input_df = user_input_df.replace(span_eng_dict)
-    elif cvt_pur_mo:
-        # change language first
-        user_input_df = user_input_df.replace(span_eng_dict)
-        # convert purpose
-        user_input_df = user_input_df.replace(map_pur_dict)
-        # convert mode
-        for a in range(len(user_input_df)):
-            if user_input_df.iloc[a]["replaced_mode"] == "same_mode":
-                # to see which row will be converted
-                logging.debug("The following rows will be changed: %s", user_input_df.iloc[a])
-                user_input_df.iloc[a]["replaced_mode"] = user_input_df.iloc[a]['mode_confirm']
+    # change language first
+    user_input_df = map_labels_sp2en(user_input_df)
+    # convert purpose
+    user_input_df = user_input_df.replace(map_pur_dict)
+    # convert mode
+    for a in range(len(user_input_df)):
+        if user_input_df.iloc[a]["replaced_mode"] == "same_mode":
+            # to see which row will be converted
+            logging.debug("The following rows will be changed: %s", user_input_df.iloc[a])
+            user_input_df.iloc[a]["replaced_mode"] = user_input_df.iloc[a]['mode_confirm']
     return user_input_df
```

Comment:
I would pull this out into a separate function.
It looks like you have three different cleanup steps.
Instead of putting them all into one or two functions, have `map_labels_sp2en` `map_labels_pur` and `map_labels_mode` separately. Then call them all in sequence in the pipeline. That makes it easier to experiment with (for example) only changing the mode or the purpose.

Reply:
done